File: Ser2023/Company/Company.Web/Modules/Company/Person/PersonColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace Company.Company.Columns;

[ColumnsScript("Person.Person")]
[BasedOnRow(typeof(PersonRow), CheckNames = true)]
public class PersonColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int BusinessEntityId { get; set; }

    //[AlignRight]
    //public int BEId {  get; set; }
    [EditLink, Width(60)]
    public string PersonType { get; set; }
    //public string Title { get; set; }
    //public string FirstName { get; set; }
    //public string MiddleName { get; set; }
    //public string LastName { get; set; }
    [EditLink, Width(250)]
    public string FullName { get; set; }

    //public string Suffix { get; set; }
}

File: 2025/Company/Company.Web/Modules/ServerTypes/Company/PersonColumns.ts
import { ColumnsBase, fieldsProxy } from '@serenity-is/corelib';
import { Column } from '@serenity-is/sleekgrid';
import { PersonRow } from './PersonRow';

export interface PersonColumns {
    BusinessEntityId: Column<PersonRow>;
    PersonType: Column<PersonRow>;
    Title: Column<PersonRow>;
    FirstName: Column<PersonRow>;
    MiddleName: Column<PersonRow>;
    LastName: Column<PersonRow>;
    Suffix: Column<PersonRow>;
    IsActive: Column<PersonRow>;
    TenantId: Column<PersonRow>;
}

export class PersonColumns extends ColumnsBase<PersonRow> {
    static readonly columnsKey = 'Company.Person';
    static readonly Fields = fieldsProxy<PersonColumns>();
}